packages/components/src/FilterBar/FilterBar.spec.tsx

```diff
@@ -91,66 +111,182 @@ const FilterBarWrapper = (
+          defaultValues={{ topping: "pearls" }}
+        />
       )
-    }
+      expect(getByRole("button", { name: "Topping : Pearls" })).toBeVisible()
+      expect(
+        getByRole("button", { name: "Remove filter - Topping" })
+      ).toBeVisible()
+    })
 
-    const { getByRole } = render(<Wrapper />)
+    it("does not show active removable filters in the Add Filters menu", async () => {
+      const { getByRole } = render(
+        <FilterBarWrapper<ValuesRemovable>
+          filters={filtersRemovable}
+          defaultValues={{ topping: "pearls" }}
+        />
+      )
 
-    const flavourButton = getByRole("button", { name: "Flavour" })
-    expect(flavourButton.textContent).toEqual("Flavour")
+      const addFiltersButton = getByRole("button", { name: "Add Filters" })
+      await user.click(addFiltersButton)
 
-    await user.click(
-      getByRole("button", { name: "Update Flavour to honey-milk-tea" })
-    )
+      const list = getByRole("list")
+      const menuOption = within(list).queryByRole("button", { name: "Topping" })
+      await waitFor(() => {
+        expect(menuOption).not.toBeInTheDocument()
+      })
+    })
+
+    it("clears the value of a filter if it is removed", async () => {
+      const { getByRole } = render(
+        <FilterBarWrapper<ValuesRemovable>
+          filters={filtersRemovable}
+          defaultValues={{ topping: "pearls" }}
+        />
+      )
+
+      const filterButton = getByRole("button", { name: "Topping : Pearls" })
+      expect(filterButton).toBeVisible()
+
+      await user.click(getByRole("button", { name: "Remove filter - Topping" }))
+      await waitFor(() => {
+        expect(filterButton).not.toBeInTheDocument()
+      })
+
+      const addFiltersButton = getByRole("button", { name: "Add Filters" })
+      await user.click(addFiltersButton)
+
+      const list = getByRole("list")
+      const menuOption = within(list).getByRole("button", { name: "Topping" })
+      await waitFor(() => {
+        expect(menuOption).toBeVisible()
+      })
+
+      await user.click(menuOption)
+      await waitFor(() => {
+        expect(list).not.toBeInTheDocument()
+      })
+      expect(getByRole("button", { name: "Topping" })).toBeVisible()
+    })
+
+    it("adds new filters in the provided order", async () => {
+      const { getByRole, getAllByTestId } = render(
+        <FilterBarWrapper<ValuesSimple>
+          filters={simpleFilters.map(filter => ({
+            ...filter,
+            isRemovable: true,
+          }))}
+        />
+      )
+
+      const addFiltersButton = getByRole("button", { name: "Add Filters" })
+      await user.click(addFiltersButton)
+
+      const menuOptionIceLevel = getByRole("button", { name: "Ice Level" })
+      await user.click(menuOptionIceLevel)
+      await user.click(addFiltersButton)
+
+      const menuOptionFlavour = getByRole("button", { name: "Flavour" })
+      await user.click(menuOptionFlavour)
+      await user.click(addFiltersButton)
```

Comment:
Looks tedious, I bet someone will want to be able to select multiple filters to bulk add at some point lol

Reply:
That's not how the UI works Geoff 😂